### PR TITLE
Feature/11881 unit input type fix

### LINKED ISSUE #11881: Project container default resource limit inputs are broken
**Describe the bug**
The unit conversion normally done by the unitinput component seems to be broken on the project create page.

**To Reproduce**
1. Navigate to the projects/namespaces list and click 'create project'
2. Navigate to the 'Container Default Resource Limit' tab
3. Type '1' in 'Cpu Reservation'

**Result**
CPU Reservation set to 1000

**Expected Result**
CPU Reservation set to 1

**Screenshots**
(I'm typing '111' in each input here)

https://github.com/user-attachments/assets/a3a2979d-902e-4d8d-a0ab-2d8b3675eca8

**Additional context**
Skipped e2e tests should be re-enabled when this is fixed.

### PR DESCRIPTION
**Pull request by cnotv**
### Summary
Fixes #11881

### Technical notes summary

Added unit tests for both component and view to ensure correct handling of this.

### Screenshot/Video

Project

https://github.com/user-attachments/assets/857a5ccb-9dac-464c-bf82-bfc8a97c7777

Namespace

https://github.com/user-attachments/assets/b83a4138-404d-424f-943f-d5453467ebc9

Workload

https://github.com/user-attachments/assets/63927508-46a2-48cd-bc67-5fa2bbb7bfa2

### Checklist
- [x] The PR is linked to an issue and the linked issue has a Milestone, or no issue is needed
- [x] The PR has a Milestone 
- [x] The PR template has been filled out
- [x] The PR has been self reviewed 
- [x] The PR has a reviewer assigned
- [x] The PR has automated tests or clear instructions for manual tests and the linked issue has appropriate QA labels, or tests are not needed
- [x] The PR has reviewed with UX and tested in light and dark mode, or there are no UX changes

### CODE REVIEW

File: shell/components/__tests__/ContainerResourceLimit.test.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import { mount } from '@vue/test-utils';
+import ContainerResourceLimit from '@shell/components/ContainerResourceLimit.vue';
+
+describe('component: ContainerResourceLimit', () => {
+  it.skip.each([
```
Comment (rak-phillip) [inline shell/components/__tests__/ContainerResourceLimit.test.ts:5]: Do we intend to skip here?

Comment (cnotv) [reply shell/components/__tests__/ContainerResourceLimit.test.ts:5]: yes, it's not rendering and I don't know why, but this test has to be there

File: shell/components/__tests__/ContainerResourceLimit.test.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    ['limitsCpu', 'cpu-reservation', '111m', '111'],
+    ['limitsMemory', 'memory-reservation', '111Mi', '111'],
+    ['requestsCpu', 'cpu-limit', '111m', '111'],
+    ['requestsMemory', 'memory-limit', '111Mi', '111'],
```
Comment (rak-phillip) [inline shell/components/__tests__/ContainerResourceLimit.test.ts:9]: It looks like the ids aren't matching. I think that we can remove the skip if we make this change

```suggestion
    ['limitsCpu', 'cpu-limit', '111m', '111'],
    ['limitsMemory', 'memory-limit', '111Mi', '111'],
    ['requestsCpu', 'cpu-reservation', '111m', '111'],
    ['requestsMemory', 'memory-reservation', '111Mi', '111'],
```

Comment (cnotv) [reply shell/components/__tests__/ContainerResourceLimit.test.ts:9]: Oh that makes sense 😅 
Thanks for noticing

**Review by rak-phillip — APPROVED** [review]
LGTM

**Pull request merged**